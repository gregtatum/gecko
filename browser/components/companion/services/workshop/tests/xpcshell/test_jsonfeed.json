{
  "license": "test components from 'CC BY-SA 3.0'-licensed https://en.wikipedia.org/wiki/JSON_Feed",
  "version": "https://jsonfeed.org/version/1.1",
  "title": "My Example Feed",
  "home_page_url": "https://example.org/",
  "feed_url": "https://example.org/feed.json",
  "description": "Optional to provide more detail beyond the title.",
  "user_comment": "Optional and should be ignored by feed readers.",
  "next_url": "https://example.org/feed-pagination.json?x=unique_next_url",
  "icon": "https://example.org/favicon-timeline-512x512.png",
  "favicon": "https://example.org/favicon-sourcelist-64x64.png",
  "authors": [
    {
      "name": "Optional Author",
      "url": "https://example.org/authors/optional-author",
      "avatar": "https://example.org/authors/optional-author/avatar-512x512.png",
      "wrong1": "it shouldn't be a part of the result",
      "wrong2": {
        "name": "hello",
        "url": "https://allizom.org"
      }
    }
  ],
  "language": "en-US",
  "items": [
    {
      "id": "2",
      "content_text": "This is a second item.",
      "url": "https://example.org/second-item",
      "language": "es-mx",
      "attachments": [
        {
          "wrong3": 123
        },
        {
          "url": "https://example.org/second-item/audio.ogg",
          "mime_type": "audio/ogg",
          "title": "Optional Title",
          "size_in_bytes": 31415927,
          "duration_in_seconds": 1800
        },
        {
          "wrong4": 456
        }
      ]
    },
    {
      "id": "required-unique-string-that-does-not-change: number, guid, url, etc.",
      "url": "https://example.org/initial-post",
      "external_url": "https://en.wikipedia.org/w/index.php?title=JSON_Feed",
      "title": "Optional Title",
      "content_html": "<p>Optional content for the feed reader. You may also use content_text or both at the same time.</p>",
      "content_text": "Optional text for simple feeds.",
      "summary": "Optional summary of the item.",
      "image": "https://example.org/initial-post/main-img.png",
      "banner_image": "https://example.org/initial-post/details-banner.png",
      "date_published": "2021-10-25T19:30:00-01:00",
      "date_modified": "2021-10-26T19:45:00-01:00",
      "authors": [
        {
          "name": "Author",
          "wrong4": "whatever"
        },
        {
          "name": "Optional Author",
          "url": "https://example.org/authors/optional-author",
          "avatar": "https://example.org/authors/optional-author/avatar-512x512.png"
        }
      ],
      "tags": [
        "Optional Tag",
        "Example"
      ],
      "language": "en-US"
    }
  ]
}
