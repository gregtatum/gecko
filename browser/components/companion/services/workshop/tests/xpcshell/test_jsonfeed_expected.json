{
  "version": "https://jsonfeed.org/version/1.1",
  "title": "My Example Feed",
  "home_page_url": "https://example.org/",
  "feed_url": "https://example.org/feed.json",
  "description": "Optional to provide more detail beyond the title.",
  "user_comment": "Optional and should be ignored by feed readers.",
  "next_url": "https://example.org/feed-pagination.json?x=unique_next_url",
  "icon": "https://example.org/favicon-timeline-512x512.png",
  "favicon": "https://example.org/favicon-sourcelist-64x64.png",
  "authors": [
    {
      "name": "Optional Author",
      "url": "https://example.org/authors/optional-author",
      "avatar": "https://example.org/authors/optional-author/avatar-512x512.png"
    }
  ],
  "language": "en-US",
  "items": [
    {
      "id": "2",
      "url": "https://example.org/second-item",
      "content_text": "This is a second item.",
      "language": "es-mx",
      "attachments": [
        {
          "url": "https://example.org/second-item/audio.ogg",
          "mime_type": "audio/ogg",
          "title": "Optional Title",
          "size_in_bytes": 31415927,
          "duration_in_seconds": 1800
        }
      ]
    },
    {
      "id": "required-unique-string-that-does-not-change: number, guid, url, etc.",
      "url": "https://example.org/initial-post",
      "external_url": "https://en.wikipedia.org/w/index.php?title=JSON_Feed",
      "title": "Optional Title",
      "content_html": "<p>Optional content for the feed reader. You may also use content_text or both at the same time.</p>",
      "content_text": "Optional text for simple feeds.",
      "summary": "Optional summary of the item.",
      "image": "https://example.org/initial-post/main-img.png",
      "banner_image": "https://example.org/initial-post/details-banner.png",
      "date_published": "2021-10-25T20:30:00.000Z",
      "date_modified": "2021-10-26T20:45:00.000Z",
      "authors": [
        { "name": "Author" },
        {
          "name": "Optional Author",
          "url": "https://example.org/authors/optional-author",
          "avatar": "https://example.org/authors/optional-author/avatar-512x512.png"
        }
      ],
      "tags": [],
      "language": "en-US"
    }
  ]
}
